applications/GeoMechanicsApplication/custom_processes/find_neighbour_elements_of_conditions_process.cpp

```diff
@@ -39,29 +39,13 @@ void FindNeighbourElementsOfConditionsProcess::Execute()
 
 void FindNeighbourElementsOfConditionsProcess::FindNeighbouringElementsForAllBoundaryTypes()
 {
-    auto generate_generic_boundaries = [](const auto& rGeometry) {
-        return rGeometry.GenerateBoundariesEntities();
-    };
-    auto generate_points   = [](const auto& rGeometry) { return rGeometry.GeneratePoints(); };
-    auto generate_edges_3d = [](const auto& rGeometry) {
-        return rGeometry.LocalSpaceDimension() == 3 ? rGeometry.GenerateEdges()
-                                                    : PointerVector<Geometry<Node>>();
-    };
-    auto generate_edges_1d = [](const auto& rGeometry) {
-        return rGeometry.LocalSpaceDimension() == 1 ? rGeometry.GenerateEdges()
-                                                    : PointerVector<Geometry<Node>>();
-    };
-
-    // Note the order in the generators: the 1D elements are only added
-    // as neighbours when the condition is not neighbouring 2D or 3D elements
-    const std::vector<std::function<PointerVector<Geometry<Node>>(const Geometry<Node>&)>> boundary_generators = {
-        generate_generic_boundaries, generate_points, generate_edges_3d, generate_edges_1d};
-
-    for (const auto& r_boundary_generator : boundary_generators) {
-        mNeighbouringEntityFinder.FindConditionNeighboursBasedOnBoundaryType(
-            r_boundary_generator, mrModelPart.Elements());
-        if (AllConditionsHaveAtLeastOneNeighbour()) return;
-    }
+    std::map<std::size_t, std::unique_ptr<BoundaryGenerator>> boundary_generator_map;
+    boundary_generator_map[0] = std::make_unique<PointsGenerator>();
+    boundary_generator_map[1] = std::make_unique<EdgesGenerator>();
+    boundary_generator_map[2] = std::make_unique<FacesGenerator>();
```

Comment:
In the unit tests, you explicitly use `std::size_t` objects rather than `int`s. Would you want to do that here, too?
```suggestion
    boundary_generator_map[std::size_t{0}] = std::make_unique<PointsGenerator>();
    boundary_generator_map[std::size_t{1}] = std::make_unique<EdgesGenerator>();
    boundary_generator_map[std::size_t{2}] = std::make_unique<FacesGenerator>();
```

Reply:
Good point, done!